### ISSUE TITLE
govuk_publishing_components unnecessarily bundles `rails` dependency

### ISSUE THREAD

**Issue by ChrisBAshton** [alphagov/govuk_publishing_components#3205]
I was wondering why the govuk-developer-docs had a rails dependency: it comes from govuk_publishing_components, which the Developer Docs only uses for a handful of components ([example](https://github.com/alphagov/govuk-developer-docs/blob/80739891478ed5af1fbab5265535042459d8ff57/source/layouts/manual_layout.html.erb#L2)).

Rails is a pretty meaty dependency and it seems a shame to include it in the gem unnecessarily. It also causes some confusion for things like the [dependency management spreadsheet](https://docs.google.com/spreadsheets/d/137KZhjctJ8qTKYPnq2QNVkyoIC6ok7KA2G1vErNC6Oo/edit#gid=1160366178), where one would be forgiven for assuming govuk-developer-docs is a rails app.

I've spiked removing rails as a hard dependency, and was still able to use components locally (in my Developer Docs build) without any rails dependency. The spike branches are below:

- https://github.com/alphagov/govuk-developer-docs/pull/3866
- https://github.com/alphagov/govuk_publishing_components/pull/3204

Making `rails` a development dependency of govuk_publishing_components (as I suspect it is needed for local development, defining routes and viewing the component locally etc) was my first step.
What complicates things is that govuk_publishing_components has a hard dependency on [govuk_personalisation](https://github.com/alphagov/govuk_personalisation), which itself [has a hard dependency on rails](https://github.com/alphagov/govuk-developer-docs/blob/5745bfe5208386f4090fbec0b28f12282e1dcd70/Gemfile.lock#L161-L169), so a second step is needed. Given that govuk_personalisation is only used [in the account layout](https://github.com/alphagov/govuk_publishing_components/pull/2289) (and the other components are otherwise untouched), I think we could do something here such as: 1) removing govuk_personalisation as a hard dependency, then 2) documenting that should anyone need to use the account layout, they need to specify govuk_personalisation as an additional dependency.

TLDR: would it be possible to decouple govuk_publishing_components from Rails? Just thinking out loud, thought I'd document in an issue.

**Comment by kevindew** [comment]
From a "not really looked into" point of view, I'm imagining Rails is pretty firmly wedged as a dependency here. Key things are:

- The Component Guide functionality, which itself is a rails engine (I'd guess there's probably a possibility that could be a separate gem or only enabled if Rails present
- The folder structure and necessary auto-loading for the templates to all operate in the Rails way. Even benignly calling for a component to render outside Rails requires loading up ActionController: https://github.com/alphagov/govuk_publishing_components/blob/c08a2d69c689bce7a457eba7b57785ed7875f8fb/lib/govuk_publishing_components.rb#L48-L61

If your goal is to remove the Rails dependency from govuk-developer-docs, I'd probably suggest that an easier direction of travel would be to utilise govuk-frontend directly in the dev docs and not use the intermediary layer of govuk_publishing_components. I imagine, assuming you only have a few component usages, that you might find the only loss is having to write some extra HTML as you'd have lost the .erb file.

**Comment by ChrisBAshton** [comment]
Thanks Kevin - you're right about my motivation. I'll explore switching Developer Docs to use govuk-frontend.
